core-web/libs/edit-content/src/lib/fields/dot-edit-content-host-folder-field/dot-edit-content-host-folder-field.component.ts

```diff
@@ -0,0 +1,129 @@
+import { createPaths } from '../../utils/functions.util';
+
+@Component({
+    selector: 'dot-edit-content-host-folder-field',
+    standalone: true,
+    imports: [TreeSelectModule, ReactiveFormsModule, TruncatePathPipe],
+    templateUrl: './dot-edit-content-host-folder-field.component.html',
+    styleUrls: ['./dot-edit-content-host-folder-field.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    viewProviders: [
+        {
+            provide: ControlContainer,
+            useFactory: () => inject(ControlContainer, { skipSelf: true })
+        }
+    ]
+})
+export class DotEditContentHostFolderFieldComponent implements OnInit {
+    @Input() field!: DotCMSContentTypeField;
+    @ViewChild(TreeSelect) treeSelect!: TreeSelect;
+    readonly #controlContainer = inject(ControlContainer);
+    readonly #editContentService = inject(DotEditContentService);
+
+    options = signal<TreeNodeItem[]>([]);
+    sitesStatus = signal<StatusRequest>('init');
+    pathControl = new FormControl();
+
+    ngOnInit() {
+        this.sitesStatus.set('loading');
+        this.#editContentService.getSitesTreePath().subscribe({
+            next: (options) => {
+                this.options.set(options);
+                this.getInitialValue();
+                this.sitesStatus.set('success');
+            },
+            error: () => {
+                this.sitesStatus.set('failed');
+            }
+        });
+    }
+
+    get formControl(): FormControl {
+        return this.#controlContainer.control.get(
+            this.field.variable
+        ) as FormControl<DotEditContentFieldSingleSelectableDataTypes>;
+    }
+
+    onNodeSelect(event: TreeNodeSelectItem) {
+        const data = event.node.data;
+        if (!data) {
+            return;
+        }
+
+        const path = `${data.hostname}:${data.path ? data.path : '/'}`;
+        this.formControl.setValue(path);
+    }
+
+    onNodeExpand(event: TreeNodeSelectItem) {
+        const { node } = event;
+        const { children, icon } = node;
+        if ((children && children.length > 0) || icon?.includes('spinner')) {
+            return;
+        }
+
+        const { hostname, path } = node.data;
+        node.icon = 'pi pi-spin pi-spinner';
+        this.#editContentService.getFoldersTreeNode(hostname, path).subscribe((children) => {
+            node.leaf = true;
+            node.icon = 'pi pi-folder-open';
+            node.children = children;
+            this.treeSelect.cd.detectChanges();
+        });
+    }
+
+    getInitialValue() {
+        const value = this.formControl.value as string;
+        if (value) {
+            const hasPaths = value.includes('/');
+            if (hasPaths) {
+                this.buildTreeByPaths(value);
+            } else {
+                const options = this.options();
+                this.pathControl.setValue(options.find((item) => item.key === value));
+            }
+        }
```

Comment:
Should this return a value? Or should we rename the function to `setInitialValue`?

```suggestion
        if (!value) {
            return;
        }

        const hasPaths = value.includes('/');
        if (hasPaths) {
            this.buildTreeByPaths(value);
            return;
        }

        const options = this.options();
        this.pathControl.setValue(options.find((item) => item.key === value));
```